metal/datasource_metal_facility_test.go

```diff
@@ -22,10 +23,35 @@ func TestAccFacilityDataSource_Basic(t *testing.T) {
 	})
 }
 
-func testAccCheckMetalFacilityDataSourceConfigBasic() string {
+func testAccDataSourceCheckMetalFacilityConfigBasic() string {
 	return `
 data "metal_facility" "test" {
     code = "ewr1"
+    desired_features = ["baremetal"]
+}
+`
+}
+
+var matchErrMissingFeature = regexp.MustCompile(`.*doesn't have feature.*`)
+
+func TestAccDataSourceFacility_Features(t *testing.T) {
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config:      testAccDataSourceCheckMetalFacilityConfigFeatures(),
+				ExpectError: matchErrMissingFeature,
+			},
+		},
+	})
+}
+
+func testAccDataSourceCheckMetalFacilityConfigFeatures() string {
+	return `
+data "metal_facility" "test" {
+    code = "ewr1"
+    desired_features = ["ibx"]
```

Comment:
This will be a very useful feature!

Reply:
I think it will be more practical to use `metro` instead of `code` or `country` in this example.

"I want any *IBX* featured facility in *{metro or address component (country, state, etc)}*" 
https://github.com/packethost/packngo/blob/v0.19.0/facilities.go#L15

We don't have client-side filters for those fields, however. Perhaps we can introduce that in a follow-up PR? (or perhaps server-side filtering supports this?)

Reply:
I wasn't thinking about the required field, `code`.

This makes sense.

Maybe someday this will be possible (https://github.com/equinix/terraform-provider-metal/pull/179/files/6b8eaf81c5b1f118b2305f5d5e54ef218e774e9d#r699831442).